### DISCUSSION TITLE
Defining number of threads as a JMeter property

### DISCUSSION THREAD

**Discussion by andreygrigoriev** [abstracta/jmeter-java-dsl#260] [Q&A] [answered]
Hi. First of all thank you for this awesome library! 

I haven't found in docs if there is a way to define the number of threads or iterations as a JMeter property like `${__P(MY_PROP)}`.
My use case is the following: I'm building a converter from Insomnia project to JMeter and as a result I want to have a JMX file that can be then parametrised

Best regards,
Andrey

**Comment by rabelenda** [comment, marked as answer]
Hello, thank your for the feedback, is really nice to hear from people enjoying the library.

In the user guide we didn't include such specific scenario, but yes, you can use them. 

You can define properties as detailed in [this user guide section](https://abstracta.github.io/jmeter-java-dsl/guide/#properties), and then reference them like this:

Eg:

```java
threadGroup()
  .rampTo("${__P(THREADS)}", Duration.Zero)
  .holdIterating("${__P(ITERATIONS)}")
  .children(
    ...
  ) 
```

We didn't include yet an option like `threadGroup("${__P(THREADS)}", "${__P(ITERATIONS)}")` since the scenario is not the most common one (from the ones we have seen) and only add to JMeterDSL class, methods for the most used scenarios (keeping the list as short as possible short, but yet including the most used features and common scenarios).

Is this enough for your use case?

**Comment by nmelnic** [comment]
This is what I put in the property file:
```
THREADS=1
ITERATIONS=1

```
**Thread group:**            
```
            threadGroup("test")
               .rampTo(Integer.parseInt("${__P(THREADS)}"),Duration.ZERO)
               .holdIterating("${__P(ITERATIONS)}")
               .children(
                  loginTransaction()
                        )
```
            
  **Error:**          
`            java.lang.NumberFormatException: For input string: "${__P(THREADS)}"`

Can someone pls advice what is wrong here?

**Reply by rabelenda** [reply]
Hello, that won't work since `Integer.parseInt` is processed by java and not Jmeter. Have you tried just using the String value in the rampTo? Eg:

```java
threadGroup("test")
               .rampTo("${__P(THREADS)}","0")
               .holdIterating("${__P(ITERATIONS)}")
               .children(
                  loginTransaction()
                        )
```
